src/Client/Error/segmentClientError.ts

```diff
@@ -0,0 +1,7 @@
+export class SegmentClientError extends Error {
+    constructor(message: string) {
+        super(message);
+        this.name = this.constructor.name; // Set the error name to the class name
```

Comment:
Might be safer to just set this as a string. Not totally sure with ESBuild but once code is minified these constructor names might change.